job-server/src/main/scala/spark/jobserver/JobServer.scala

```diff
@@ -90,6 +97,36 @@ object JobServer {
     }
   }
 
+  private def maskCredentials(lines: String, credentialRegex: Regex): String = {
+    lines
+      .split("\n")
+      .toSeq
+      .map {
+        line => line.split(":") match {
+          // if key matches credential keys pattern and value is not empty, mask credentials
+          case Array(key, value) if (credentialRegex.findFirstIn(key).nonEmpty
+            && EMPTY_VALUE_PATTERN.findFirstIn(value.stripMargin).isEmpty) =>
+            Array(key, CREDENTIAL_MASK).mkString(":")
+          case _ => line
+      }
+    }.mkString("\n")
+  }
+
+  private def credentialRegex(config: Config): Regex = {
+    // Use default credential keys if spark.ui.confidentialKeys is not set
+    val credentialKeys = try {
+      config.getString("spark.ui.confidentialKeys")
+    } catch {
+      case _: Exception =>
+        logger.info(s"spark.ui.confidentialKeys is not set, " +
+          s"use default credential keys $DEFAULT_CREDENTIAL_KEYS")
+        DEFAULT_CREDENTIAL_KEYS
+    }
+
+    // Case insensitive
+    s"""(?i)(${credentialKeys.split(",").mkString("|")})" """.r
```

Comment:
why space at the end?

Reply:
There is space at the end after string split